src/Adapter/Product/Validate/ProductValidator.php

```diff
@@ -133,6 +149,101 @@ private function validateShipping(Product $product): void
         $this->validateProductLocalizedProperty($product, 'delivery_out_stock', ProductConstraintException::INVALID_DELIVERY_TIME_OUT_OF_STOCK_NOTES);
     }
 
+    /**
+     * @param Product $product
+     *
+     * @throws ProductConstraintException
+     * @throws ProductPackConstraintException
+     * @throws ProductStockConstraintException
+     */
+    private function validateStock(Product $product): void
+    {
+        $advancedStockEnabled = (bool) $this->configuration->get('PS_ADVANCED_STOCK_MANAGEMENT');
```

Comment:
Maybe we could extract this property in constructor to store only the property, not the Configuration service ?

Reply:
```suggestion
        $advancedStockEnabled = $this->configuration->getBoolean('PS_ADVANCED_STOCK_MANAGEMENT');
```

Reply:
Ok I did it (along with the default pack stock type) However this forces us to reboot the kernel for each feature because the configuration is no longer dynamic on runtime It is injected once when the the service is created and can not change later

For most use cases it's acceptable, but it also means that switching this parameter in the BO requires to clean the cache We must keep this in mind for other cases with configuration that are more likely to be changed by the merchant

Reply:
Actually it doesn't require to clean the cache, my bad Since the DI gets the value from database at each new process it's ok But we still should think a bit about this when we use configuration in Dependency Injection